File: writer-reader-backend/app/Models/Message.php
<?php

namespace App\Models;

use Database\Factories\WorkFactory;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * @class Message
 * @package App\Models
 * @property string id
 * @property string content
 * @property int sent_by_id
 * @property int sent_to_id
 * @property string created_at
 * @property string updated_at
 * @property User sender
 * @property User receiver
 */
class Message extends Model
{
    /** @use HasFactory<WorkFactory> */
    use HasFactory;
    use HasUuids;

    protected $fillable = [
        'content',
        'sent_by_id',
        'sent_to_id',
        'created_at',
        'updated_at',
    ];

    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function sender(): BelongsTo
    {
        return $this->belongsTo(User::class, 'sent_by_id');
    }

    public function receiver(): BelongsTo
    {
        return $this->belongsTo(User::class, 'sent_to_id');
    }
}

File: writer-reader-android-client/app/src/main/java/hu/bme/aut/android/writer_reader_client/data/model/get/Message.kt
package hu.bme.aut.android.writer_reader_client.data.model.get

import com.squareup.moshi.Json

data class Message(
    val id: String,
    val content: String,
    @Json(name = "sent_by_id") val sentById: String,
    @Json(name = "sent_to_id") val sentToId: String,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "updated_at") val updatedAt: String
)